overwatch-rs/src/services/handle.rs

```diff
@@ -15,52 +15,60 @@ use crate::services::{ServiceCore, ServiceData, ServiceId, ServiceState};
 // unnecessary Option and cloning.
 /// Service handle
 /// This is used to access different parts of the service
-pub struct ServiceHandle<S: ServiceData> {
+pub struct ServiceHandle<Message, Settings, Data, State> {
     /// Message channel relay
     /// Would be None if service is not running
     /// Will contain the channel if service is running
-    outbound_relay: Option<OutboundRelay<S::Message>>,
+    outbound_relay: Option<OutboundRelay<Message>>,
     /// Handle to overwatch
     overwatch_handle: OverwatchHandle,
-    settings: SettingsUpdater<S::Settings>,
-    status: StatusHandle<S>,
-    initial_state: S::State,
+    settings: SettingsUpdater<Settings>,
+    status: StatusHandle<Data>,
+    initial_state: State,
 }
 
 /// Service core resources
 /// It contains whatever is necessary to start a new service runner
-pub struct ServiceStateHandle<S: ServiceData> {
+pub struct ServiceStateHandle<Message, Settings, Data, State> {
     /// Relay channel to communicate with the service runner
-    pub inbound_relay: InboundRelay<S::Message>,
-    pub status_handle: StatusHandle<S>,
+    pub inbound_relay: InboundRelay<Message>,
+    pub status_handle: StatusHandle<Data>,
```

Comment:
Should this be `Status` instead of `Data`?

Reply:
Yes good catch, I was mislead by the previous bounds on the `StatusHandle` type. I will refrain from pushing stuff now until other people have looked at this, but I will fix it later, thanks.